setup.py

```diff
@@ -0,0 +1,42 @@
+import os
+import re
+
+from setuptools import setup
+from setuptools_scm import Configuration, _do_parse
+
+
+def get_version() -> str:
+    """Create version string based on branch name.
+
+    release branch -> 1.2.3rc4
+    develop branch -> 1.2.3a4
+    other branch -> 1.2.3.dev4+<branch_name
```

Comment:
nit: `>` at the end.
Maybe include couple of real examples where chars become `.`s?

Reply:
updated